python/cudf_polars/tests/dsl/test_serialization.py

```diff
@@ -0,0 +1,68 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import pickle
+
+import pytest
+
+from polars.polars import _expr_nodes as pl_expr
+
+from cudf_polars.dsl.expressions.boolean import BooleanFunction
+from cudf_polars.dsl.expressions.datetime import TemporalFunction
+from cudf_polars.dsl.expressions.string import StringFunction
+
+
+@pytest.mark.parametrize(
+    "function", [BooleanFunction, TemporalFunction, StringFunction]
+)
+def test_function_name_serialization_all_values(function):
+    # Test serialization and deserialization for all values of function.Name
+    for name in function.Name:
+        serialized_name = pickle.dumps(name)
+        deserialized_name = pickle.loads(serialized_name)
+        assert deserialized_name is name
+
+
+@pytest.mark.parametrize(
+    "function", [BooleanFunction, TemporalFunction, StringFunction]
+)
+def test_function_name_invalid(function):
+    # Test invalid attribute name
+    with pytest.raises(
+        AttributeError, match="type object 'Name' has no attribute 'InvalidAttribute'"
+    ):
+        assert function.Name.InvalidAttribute is function.Name.InvalidAttribute
+
+
+@pytest.mark.parametrize(
+    "function", [BooleanFunction, TemporalFunction, StringFunction]
+)
+def test_from_polars_all_names(function):
+    # Test that all valid names of polars expressions are correctly converted
+    for name in function.Name:
+        polars_function = getattr(pl_expr, function.__name__)
+        polars_function_attr = getattr(polars_function, name.name)
+        cudf_function = function.Name.from_polars(polars_function_attr)
+        assert cudf_function == name
```

Comment:
This tests that all the names we state match existing polars names. But not the converse, that all names that polars has match ones we know about.

How about:
```
polars_function = getattr(pl_expr, function.__name__)
polars_names = [name for name in dir(polars_function) if not name.startswith("_")]
# Check names advertised by polars are the same as we advertise
assert set(polars_names) == set(function.Name.__members__)
for name in function.Name:
    attr = getattr(polars_function, name)
    assert function.Name.from_polars(attr) == name
```

Reply:
This is indeed a better idea, applied changes in https://github.com/rapidsai/cudf/pull/17418/commits/4de3b79e92477d8595d28d07155203d0ea7f61db .